### PR TITLE
SLIM-682 interval meter reads

### PR DESCRIPTION
**Pull request by rudidevries**

### CODE REVIEW

File: integration-tests-cucumber/src/test/resources/SmartMeteringMonitoring.feature [hunk 1]
```diff
@@ -22,15 +22,18 @@ Feature:
...
-    Then the periodic meter reads gas result should be returned
+    Then the "<periodType>" meter reads gas result should be returned
 
-  @SLIM-228 @SmartMeterMonitoring
-  Scenario: Get the interval meter reads from a gas device
-    When the get interval meter reads gas request is received
-    Then the interval meter reads gas result should be returned
+    Examples:
+      |periodType	|beginDate	|endDate	|
+      |INTERVAL		|2015-09-01	|2015-10-01	|
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/resources/SmartMeteringMonitoring.feature]: Please fix the alignment.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java [hunk 2]
```diff
@@ -54,44 +53,37 @@
...
+        PROPERTIES_MAP.put(PERIOD_TYPE_LABEL, arg1);
     }
 
-    @Then("^the periodic meter reads gas result should be returned$")
-    public void thePeriodicMeterReadsResultShouldBeReturned() throws Throwable {
-        PROPERTIES_MAP.put(CORRELATION_UID_LABEL, this.correlationUid);
-        this.responseRunner(PROPERTIES_MAP, TEST_CASE_NAME_RESPONSE, LOGGER);
+    @Given("^the begin date \"([^\"]*)\"$")
+    public void theBeginDate(final String arg1) {
+        PROPERTIES_MAP.put(BEGIN_DATE_LABEL, arg1);
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: Could you use a more descriptive variable name than 'arg1' ? Maybe something like: 'beginDate'.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java [hunk 3]
```diff
@@ -54,44 +53,37 @@
...
-    public void theGetPeriodicMeterReadsRequestIsReceived() throws Throwable {
-
-        PROPERTIES_MAP.put(DEVICE_IDENTIFICATION_G_LABEL, this.deviceId.getDeviceIdG());
-        PROPERTIES_MAP.put(ORGANISATION_IDENTIFICATION_LABEL, this.organisationId.getOrganisationId());
-        PROPERTIES_MAP.put(ENDPOINT_LABEL, this.serviceEndpoint.getServiceEndpoint());
-
-        this.requestRunner(TestStepStatus.OK, PROPERTIES_MAP, TEST_CASE_NAME_REQUEST, TEST_CASE_XML_225, TEST_SUITE_XML);
+    @Given("^the period type \"([^\"]*)\"$")
+    public void thePeriodType(final String arg1) {
+        PROPERTIES_MAP.put(PERIOD_TYPE_LABEL, arg1);
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: Could you use a more descriptive variable name than 'arg1' ? Maybe something like: 'periodType'.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java [hunk 4]
```diff
@@ -54,44 +53,37 @@
...
+    }
 
-        assertTrue(this.runXpathResult.assertXpath(this.response, PATH_RESULT_PERIODTYPE,
-                XPATH_MATCHER_RESULT_PERIODTYPE));
-        assertTrue(this.runXpathResult.assertXpath(this.response, PATH_RESULT_LOGTIME, XPATH_MATCHER_RESULT_LOGTIME));
-        assertTrue(this.runXpathResult.assertXpath(this.response, PATH_RESULT_CONSUMPTION, XPATH_MATCHER_RESULT));
-        assertTrue(this.runXpathResult.assertXpath(this.response, PATH_RESULT_CAPTURETIME, XPATH_MATCHER_RESULT));
+    @Given("^the end date \"([^\"]*)\"$")
+    public void theEndDate(final String arg1) {
+        PROPERTIES_MAP.put(END_DATE_LABEL, arg1);
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: Could you use a more descriptive variable name than 'arg1' ? Maybe something like: 'endDate'.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java [hunk 5]
```diff
@@ -54,44 +53,37 @@
...
         PROPERTIES_MAP.put(ENDPOINT_LABEL, this.serviceEndpoint.getServiceEndpoint());
 
-        this.requestRunner(TestStepStatus.OK, PROPERTIES_MAP, TEST_CASE_NAME_REQUEST, TEST_CASE_XML_228, TEST_SUITE_XML);
+        this.requestRunner(TestStepStatus.OK, PROPERTIES_MAP, TEST_CASE_NAME_REQUEST, TEST_CASE_XML, TEST_SUITE_XML);
     }
 
-    @Then("^the interval meter reads gas result should be returned$")
-    public void theIntervalMeterReadsResultShouldBeReturned() throws Throwable {
+    @Then("^the \"([^\"]*)\" meter reads gas result should be returned$")
+    public void theMeterReadsGasResultShouldBeReturned(final String arg1) throws Throwable {
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: Could you use a more descriptive variable name than 'arg1' ? Maybe something like: 'periodType'.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/RunCukesTest.java [hunk 6]
```diff
@@ -16,9 +16,9 @@
 import cucumber.api.junit.Cucumber;
 
 @RunWith(Cucumber.class)
-@CucumberOptions(features = "src/test/resources/", tags = { "~@SLIM-216", "~@SLIM-256", "~@SLIM-261", "~@SLIM-228",
-        "~@SLIM-484", "~@SLIM-501", "~@SLIM-150", "~@SLIM-441" }, plugin = { "pretty",
-        "html:target/output/Cucumber-report", "html:target/output/Cucumber-html-report.html" }, snippets = SnippetType.CAMELCASE)
+@CucumberOptions(features = "src/test/resources/", tags = { "~@SLIM-256", "~@SLIM-261", "~@SLIM-484", "~@SLIM-501",
```
Comment (riannefijn) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/RunCukesTest.java]: Why remove the ~@SLIM-216 tag? Not only here, but everywhere?

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java (hunk 2 above)
Comment (rudidevries) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/smartmeteringmonitoring/PeriodicMeterReadsGas.java]: The joys of generated code :-) Will change it.

File: integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/RunCukesTest.java (hunk 6 above)
Comment (rudidevries) [inline integration-tests-cucumber/src/test/java/com/alliander/osgp/platform/cucumber/RunCukesTest.java]: The test for SLIM-216 did not work, but will work now the simulator has been fixed. So this exclusion can be removed.

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://52.16.237.165:80/OSGP/Integration-Tests?pullRequest=57) <br/>All committers of the pull request should sign our Contributor License Agreement in order to get your pull request merged.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: frankheijnders<br/>:white_check_mark: rudidevries<br/>:x: riannefijn

**Pull request merged**